src/web/mjs/connectors/YoungChampion.mjs

```diff
@@ -0,0 +1,139 @@
+                this.defaultOrder.push([i, j]);
+            }
+        }
+    }
+
+    async _getMangaFromURI(uri) {
+        const request = new Request(uri);
+        const [data] = await this.fetchDOM(request, 'h1.series-h-title span:not([class])');
+        const id = uri.pathname;
+        const title = data.textContent.trim();
+        return new Manga(this, id, title);
+    }
+
+    async _getMangas() {
+        let msg = 'This website does not provide a manga list, please copy and paste the URL containing the chapters directly from your browser into HakuNeko.';
+        throw new Error(msg);
+    }
+
+    async _getChapters(manga) {
+        const uri = new URL(manga.id+'/list', this.url);
+        const request = new Request(uri);
+        const data = await this.fetchDOM(request, 'div.series-ep-list a');
+        return data.map(chapter => {
+            return {
+                id : new URL(chapter.dataset['href']).pathname,
+                title : chapter.querySelector('span.series-ep-list-item-h-text').textContent.trim()
+            };
+        });
+    }
+
+    async _getPages(chapter) {
+        const uri = new URL(chapter.id, this.url);
+        const request = new Request(uri);
+        const [viewer] = await this.fetchDOM(request, '#comici-viewer');
+        if (!viewer) {
+            throw new Error(`The chapter '${chapter.title}' is neither public, nor purchased!`);
+        }
+        const coord = await this._fetchCoordInfo(viewer);
+        return coord.result.map(image => {
+            return this.createConnectorURI({
+                url: image.imageUrl,
+                scramble: image.scramble,
+            });
+        });
+    }
+
+    async _handleConnectorURI(payload) {
+        const request = new Request(new URL(payload.url));
+        request.headers.set('x-origin', this.url);
+        request.headers.set('x-referer', this.url);
+        const res = await fetch(request);
+        const blob = await res.blob();
+        const image = await createImageBitmap(blob);
+        const canvas = this._descramble(image, payload.scramble);
+        const blobFinally = await this._canvasToBlob(canvas);
+        return this._blobToBuffer(blobFinally);
+    }
+
+    async _fetchCoordInfo(viewer) {
+        //first request get page count
+        let uri = new URL('/book/contentsInfo', this.url);
+        uri.searchParams.set('comici-viewer-id', viewer.getAttribute('comici-viewer-id'));
+        uri.searchParams.set('user-id', viewer.dataset['memberJwt']);
+        uri.searchParams.set('page-from', '0');
+        uri.searchParams.set('page-to', '1');
+        let request = new Request(uri);
+        request.headers.set('x-origin', this.url);
+        request.headers.set('x-referer', this.url);
+        const data = await this.fetchJSON(request);
+
+        //second request fetch actual pages data
+        const numbers = data.totalPages;
+        uri = new URL('/book/contentsInfo', this.url);
```

Comment:
Seems to be the same URL as used previously, consider re-using `uri`, but just update the `page-to` search parameter